presto-main/src/main/java/com/facebook/presto/operator/window/AggregateWindowFunction.java

```diff
@@ -80,15 +80,20 @@ else if ((frameStart == currentStart) && (frameEnd >= currentEnd)) {
 
     private void accumulate(int start, int end)
     {
-        // TODO: add Accumulator method that does not require creating pages
-        PageBuilder pageBuilder = new PageBuilder(function.getParameterTypes());
-        for (int position = start; position <= end; position++) {
-            for (int i = 0; i < function.getParameterTypes().size(); i++) {
-                windowIndex.appendTo(argumentChannels[i], position, pageBuilder.getBlockBuilder(i));
+        if (function.getParameterTypes().size() == 0) {
+            PageBuilder pageBuilder = new PageBuilder(function.getParameterTypes());
+            for (int position = start; position <= end; position++) {
+                pageBuilder.declarePosition();
+            }
+            accumulator.addInput(pageBuilder.build());
+        }
+        else {
+            for (int position = start; position <= end; ++position) {
+                for (int i = 0; i < function.getParameterTypes().size(); ++i) {
+                    accumulator.addInput(windowIndex, argumentChannels[i], position);
```

Comment:
Rather than having a loop here, which could create a bunch of intermediate blocks, why not do a single call with the start/end?